packages/assets-controllers/src/NftController.ts

```diff
@@ -1189,6 +1189,13 @@ export class NftController extends BaseControllerV1<NftConfig, NftState> {
           openSeaEnabled,
           isIpfsGatewayEnabled,
         });
+
+        const needsUpdateNftMetadata =
+          (isIpfsGatewayEnabled && ipfsGateway !== '') || openSeaEnabled;
+
+        if (needsUpdateNftMetadata) {
+          this.updateNftMetadata({ userAddress: selectedAddress });
+        }
```

Comment:
Initially i wanted to have a logic here where i only make the update when the previous value of the toggle is "OFF" and the new value is "ON"; so it looked something like this

```
        const {
          openSeaEnabled: previousOpenseaEnabled,
          isIpfsGatewayEnabled: previousIpfsEnabled,
        } = this.config;
        const needsUpdateNftMetadata =
          (!previousIpfsEnabled &&
            isIpfsGatewayEnabled &&
            ipfsGateway !== '') ||
          (!previousOpenseaEnabled && openSeaEnabled);
```
And then i update 
`        if (needsUpdateNftMetadata) {
          await this.updateNftMetadata({ userAddress: selectedAddress });
        }`

The "this.configure" here is being called twice in the case where the user toggles on and off the IPFS toggle.
The first time its called is to set the values of the toggles
The second time its called is to set the ipfsGateway => ipfsGateway="default value or user's value" 

Which means that trying to compare the previous value with the current value wont be working correctly because the second time the preferenceState changes to set the ipfs gateway, the previousIpfsEnabled will be already changed to isIpfsGatewayEnabled.
Eventually i thought there might be little value in trying to make this logic;
1- First scenario (happy scenario): where the toggles are initially OFF then user imports an NFT then turns "display media" ON we will update nft metadata.
2- Second scenario which will be avoided by this logic, is when toggles are ON then the user toggles IPFS OFF, then we do not call updateNftMetadata. However, with the logic pushed, we are filtering only nfts that do not have name/image/discription, so if all NFTs have already those attributes we wont fetch metadata and if some of them are still missing those attributes for any reason, we will attempt to refetch again(which is the only tradeoff here).